### PR TITLE
Transition to search_teaching_events_grouped_by_type

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-649](https://trello.com/c/yBEK1KXr/649-dev-refactor-search-indexed-by-endpoints-to-have-a-better-response)

### Context

We currently use the `search_teaching_events_indexed_by_type` and `upcoming_teaching_events_indexed_by_type` queries, which return in the format:

```
{
  "<type_id>": [... events ...],
  ...
}
```

Instead, the API now has a new `search_teaching_events_grouped_by_type` endpoint that returns results in a more rest-ful format:

```
[
  { "typeId": <type_id>, "teachingEvents": [... events ...] },
   ...
]
```

Note the upcoming endpoint has not been udpated, and instead we are removing it completely as it can be achieved by passing a `startAfter` parameter to the search method.

All occurances of calls to `..._indexed_by_..` have been updated.

### Changes proposed in this pull request

- Transition to search_teaching_events_grouped_by_type

- Bump ApiClient version

Pulls in the new `search_teaching_events_grouped_by_type` method.

### Guidance to review

I've given this a decent test locally to ensure all the event searching etc works as expected.

### CODE REVIEW

File: spec/support/helpers/events.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Helpers
```
Comment (jebw) [inline spec/support/helpers/events.rb:1]: This is already a top level module in rails, and _I think_ will include automatically include all modules below it in Controllers and Views - potentially causing issues for specs in the future.

I'd suggest renaming this to something like `SpecHelpers` ?

**Review by jebw — CHANGES_REQUESTED** [review]
LGTM apart from the use of `Helpers` module in the `spec/support` space :+1:

File: spec/support/helpers/events.rb (hunk 1 above)
Comment (ethax-ross) [reply spec/support/helpers/events.rb:1]: Oh ok, I was going off the RSpec docs but happy to rename it:

https://relishapp.com/rspec/rspec-core/docs/helper-methods/define-helper-methods-in-a-module

**Review by jebw — APPROVED** [review]
LGTM

**Pull request merged**